src/scroll-timeline-css-parser.js

```diff
@@ -127,17 +135,27 @@ export class StyleParser {
     return null;
   }
 
-  getViewTimelineOptions(timelineName, target) {
+  concatAnimationDelays(animOptions) {
+    const animationDelay = animOptions['animation-delay'] ? animOptions['animation-delay'] :
+      defaultAnimationDelay();
+    const animationEndDelay = animOptions['animation-end-delay'] ? animOptions['animation-end-delay'] :
+      defaultAnimationEndDelay();
+
+    return `${animationDelay} ${animationEndDelay}`;
+  }
+
+  getViewTimelineOptions(timelineName, animOptions) {
     // TODO: Take into account the scoping of the ViewTimelines
     // https://github.com/w3c/csswg-drafts/issues/7047
     for (let i = this.subjectSelectorToViewTimeline.length - 1; i >= 0; i--) {
       const options = this.subjectSelectorToViewTimeline[i];
       if(options.name == timelineName) {
-        const allSubjects = target.parentElement.querySelectorAll(options.selector);
+        const allSubjects = document.querySelectorAll(options.selector);
         if(allSubjects.length) {
           return {
             subject: allSubjects[allSubjects.length - 1],
-            axis: options.axis
+            axis: options.axis,
+            timeRange: parseTimeRange(this.concatAnimationDelays(animOptions))
```

Comment:
Multiple animations can share the same timeline but have different timeline ranges.